pkg/route/graph/analysis/analysis.go

```diff
@@ -70,6 +75,54 @@ route:
 
 				continue route
 			}
+
+			if routeNode.Spec.Port != nil {
+				routePortString := routeNode.Spec.Port.TargetPort.String()
+				if routePort, err := strconv.Atoi(routePortString); err == nil {
+					// route has numeric port, service has no port with that number as a targetPort.
+					for _, port := range svcNode.Spec.Ports {
+						if port.TargetPort.IntValue() == routePort {
+							continue route
```

Comment:
named `continue` targets smell of lots of cyclomatic complexity, maybe we'd be better off with breaking this function up and getting rid of the need for them